graphene/types/objecttype.py

```diff
@@ -14,7 +19,34 @@ class ObjectTypeOptions(BaseOptions):
     interfaces = ()  # type: Iterable[Type[Interface]]
 
 
-class ObjectType(BaseType):
+class ObjectTypeMeta(BaseTypeMeta):
+    def __new__(cls, name, bases, namespace):
+        # We create this type, to then overload it with the dataclass attrs
+        class InterObjectType:
+            pass
+
+        base_cls = super().__new__(cls, name, (InterObjectType,) + bases, namespace)
+        if base_cls._meta:
+            fields = [
+                (
+                    key,
+                    "typing.Any",
```

Comment:
Can this typing be more accurate by taking the type of the GraphQL field?

Reply:
Yes, that would be ideal. However I have a bigger plan that we can action in a separate PR.